modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectRelationship/ObjectRelationshipInheritanceCheckbox.tsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayCheckbox} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayPopover from '@clayui/popover';
+import React, {useState} from 'react';
+
+import './ObjectRelationshipInheritanceCheckbox.scss';
+
+interface ObjectRelationshipInheritanceCheckbox {
+	onSubmit: (values?: Partial<ObjectRelationship>) => Promise<void>;
+	setValues: (values: Partial<ObjectRelationship>) => void;
+	values: Partial<ObjectRelationship>;
+}
+
+export function ObjectRelationshipInheritanceCheckbox({
+	onSubmit,
+	setValues,
+	values,
+}: ObjectRelationshipInheritanceCheckbox) {
+	const [showPopover, setShowPopover] = useState(false);
+
+	return (
+		<>
+			<div className="form-group lfr__object-relationship-inheritance-container">
+				<ClayCheckbox
+					checked={!!values.edge}
+					label={Liferay.Language.get(
+						'enable-inheritance-including-shared-permissions-and-grouped-apis'
+					)}
+					onChange={({target}) => {
+						if (target.checked) {
+							setValues({
+								...values,
+								edge: true,
+							});
+						}
+						else {
+							const parentWindow = Liferay.Util.getOpener();
+
+							parentWindow.Liferay.fire(
+								'openModalDisableInheritance',
+								{
+									handleDisable: async () => {
```

Comment:
This is a simple function call, we have no complexity here and no code repetition.  I added a comment below regarding the removal of the setValues ​​call. This will make the function even simpler.

I don't think we will need to move the onSubmit call to another function. Let me know what you think about this.